src/modules/user/user.service.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
+import { InjectRepository } from '@nestjs/typeorm'
+import { Repository } from 'typeorm'
+import { UserEntity } from './entities/user.entity'
+import { User } from './domain/user'
+import { UserUpdate } from './domain/user-update.domain'
+import { UserCreate } from './domain/user-create.domain'
+
+@Injectable()
+export class UserService {
+  constructor(
+    @InjectRepository(UserEntity)
+    private readonly userRepository: Repository<UserEntity>,
+  ) {}
+
+  async isEmailExisting(email: string): Promise<boolean> {
+    const count = await this.userRepository.count({ where: { email } })
+    return count > 0
+  }
+
+  async create(userCreate: UserCreate): Promise<User> {
+    const emailExists = await this.isEmailExisting(userCreate.email)
+    if (emailExists) {
+      throw new BadRequestException('Email already exists')
+    }
+    const userEntity = await this.userRepository.save(userCreate.toEntity())
+    return User.fromEntity(userEntity)
+  }
+
+  async findAll(): Promise<User[]> {
+    const userEntitys = await this.userRepository.find()
+    if (userEntitys.length === 0) {
+      return []
+    }
+    return userEntitys.map((user) => User.fromEntity(user))
```

Comment:
Đưa hàm vào class user


```suggestion
    return User.fromEntities(userEntities);
```